Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +20,150 @@
+
+    @Override
+    public ITransformedNetwork getTransformedNetwork(String networkId) throws QuadrigaStorageException {
+
+        ITransformedNetwork networkJSon = null;
+
+        List<INetworkNodeInfo> networkTopNodesList = null;
+
+        try {
+            networkTopNodesList = networkManager.getNetworkTopNodes(networkId);
+        } catch (QuadrigaStorageException e) {
+            logger.error("DB Error while getting network top nodes", e);
+            return null;
+        }
+
+        networkJSon = transformer.transformNetwork(networkTopNodesList);
+
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork getTransformedNetworkOfProject(String projectId)
+            throws QuadrigaStorageException {
+
+        List<INetwork> networkList = getNetworkList(projectId);
+        if (networkList == null) {
+            return null;
+        }
+
+        List<INetworkNodeInfo> networkNodeInfoList = new ArrayList<INetworkNodeInfo>();
+        for (INetwork network : networkList) {
+            try {
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                if (localNetworkNodeInfoList != null) {
+                    networkNodeInfoList.addAll(localNetworkNodeInfoList);
+                }
+            } catch (QuadrigaStorageException e) {
+                throw new QuadrigaStorageException("Database Error while getting network top nodes for a network with ID: " +
+                        network.getNetworkId(), e);
+            }
+        }
+
+        ITransformedNetwork networkJSon = transformer.transformNetwork(networkNodeInfoList);
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork searchTransformedNetworkOfProject(String projectId, String conceptId)
+            throws QuadrigaStorageException {
+        List<INetwork> networkList = getNetworkList(projectId);
+        if (networkList == null) {
+            return null;
+        }
+
+        Map<String, Node> searchNodes = new HashMap<String, Node>();
+        List<Link> searchLinks = new ArrayList<Link>();
+        for (INetwork network : networkList) {
+            try {
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                // check if this network has the node with given concept id
+                if (localNetworkNodeInfoList != null) {
+                    ITransformedNetwork transformedNetwork =
+                            transformer.transformNetwork(localNetworkNodeInfoList);
+                    // search for concept id
+                    Node searchNode = null;
+                    for (Node node: transformedNetwork.getNodes().values()) {
+                        if (node.getConceptId().equals(conceptId)) {
+                            searchNode = node;
+                            break;
```

Comment:
hm, so wouldn't that get only one node with the given concept form each network? But if I search for a predicate (e.g. engages with), I might have several nodes with that concept. Or am I missing something?